src/ctype/CTypeSchema.ts

```diff
@@ -105,19 +108,22 @@ export const CTypeModel = {
           properties: {
             type: {
               type: 'string',
-              enum: ['string', 'integer', 'number', 'boolean'],
+              enum: ['string', 'integer', 'number', 'boolean', 'object'],
+            },
+            $ref: {
+              type: 'string',
+              format: 'uri-reference',
             },
             format: {
               type: 'string',
-              enum: ['date', 'time', 'uri'],
+              enum: ['date', 'time', 'uri-reference'],
             },
           },
-          required: ['type'],
```

Comment:
@tjwelde I had to remove this due to the `ajv` checking the `type`. If it had both a `$ref` and `type` it would use the `type` to validate over the `$ref` 

Maybe, something I could do differently.

Reply:
I fixed it with the `oneOf`

Reply:
NICE